designer/server/src/main/scala/pl/touk/nussknacker/ui/api/DefinitionResources.scala

```diff
@@ -37,6 +40,18 @@ class DefinitionResources(modelDataProvider: ProcessingTypeDataProvider[ModelDat
           modelDataProvider.mapValues(_.processDefinition.services.mapValuesNow(UIProcessObjectsFactory.createUIObjectDefinition(_, processCategoryService))).all
         }
       }
+    } ~ path("processDefinitionData" / "categoriesWithProcessingType") {
+      get {
+        complete {
+          val categories = user match {
+            case user: CommonUser => processCategoryService.getAllCategories.filter(user.can(_, Read))
```

Comment:
why not `processCategoryService.getUserCategories` ?

Reply:
Didn't noticed that so I reinvented the wheel .. Thanks, changed.